api/envoy/type/format.proto

```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+
+package envoy.type;
+
+option java_outer_classname = "FormatProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.type";
+
+import "google/protobuf/struct.proto";
+
+message StringOrJson {
+  // Allows to define custom format of returned data.
+  oneof format {
+    // Plain text format.
```

Comment:
In the final version of the PR can you flesh these docs out to cross-link to the format rules, etc.? It's not super clear how the user is supposed to fill these fields out. Thank you!

/wait